src/main/scala/tilelink/FIFOFixer.scala

```diff
@@ -98,6 +100,31 @@ class TLFIFOFixer(policy: TLFIFOFixer.Policy = TLFIFOFixer.all)(implicit p: Para
         out.c.valid := Bool(false)
         out.e.valid := Bool(false)
       }
+
+//Functional cover properties
+     
+      cover(in.a.valid && stall, "COVER FIFOFIXER STALL", "Cover: Stall occured for a valid transaction")
+
+      val SourceIdFIFOed = RegInit(UInt(0, width = edgeIn.client.endSourceId))
+      val SourceIdSet = Wire(init = UInt(0, width = edgeIn.client.endSourceId))
+      val SourceIdClear = Wire(init = UInt(0, width = edgeIn.client.endSourceId))
+
+      when (a_first && in.a.fire() && !a_notFIFO)  {
+        SourceIdSet := UIntToOH(in.a.bits.source)
+      }
+      when (d_first && in.d.fire())  {
+        SourceIdClear := UIntToOH(in.d.bits.source)
+      }
+
+      SourceIdFIFOed := SourceIdFIFOed | SourceIdSet
+      val allIDs_FIFOed = SourceIdFIFOed===Fill(SourceIdFIFOed.getWidth, 1.U)
+
+      cover(allIDs_FIFOed, "COVER all sources", "Cover: FIFOFIXER covers all Source IDs")
```

Comment:
Why don't you just do the following instead of having a tracking variable
(0 to edgeIn.client.endSourceId).foreach { id =>
  cover(a_first && in.a.fire() && !a_notFIFO && in.a.bits.source===id.U, s"Cover: FIFOFIXER covers id ${id}")
}

Reply:
I have no idea what is the purpose of `cove`? Can you clarify this? Is this way of doing code coverage? Is there some info I can get about the cover stuff? Is it a framework? Thanks!